AutomatticTracks/src/main/java/com/automattic/android/tracks/CrashLogging/CrashLogging.kt

```
@@ -0,0 +1,109 @@
+package com.automattic.android.tracks.crashlogging
+
+import android.content.Context
+import com.automattic.android.tracks.BuildConfig
+import io.sentry.NoOpLogger
+import io.sentry.Sentry
+import io.sentry.SentryEvent
+import io.sentry.SentryLevel
+import io.sentry.SentryOptions
+import io.sentry.SystemOutLogger
+import io.sentry.android.core.SentryAndroid
+import io.sentry.protocol.Message
+import io.sentry.protocol.User
+import java.util.Locale
+
+object CrashLogging {
+
+    private const val EMPTY_VALUE = ""
+
+    private lateinit var dataProvider: CrashLoggingDataProvider
+
+    @JvmStatic
+    fun start(
+        context: Context,
+        dataProvider: CrashLoggingDataProvider,
+    ) {
+        SentryAndroid.init(context) { options ->
+            options.apply {
+                dsn = dataProvider.sentryDSN
+                environment = dataProvider.buildType
+                release = dataProvider.releaseName
+                setDebug(BuildConfig.DEBUG)
+                setLogger(if (BuildConfig.DEBUG) SystemOutLogger() else NoOpLogger.getInstance())
+                setTag("locale", getCurrentLanguage(dataProvider.locale))
+                beforeSend = SentryOptions.BeforeSendCallback { event, _ ->
+                    return@BeforeSendCallback if (dataProvider.userHasOptedOut) {
+                        null
+                    } else {
+                        event
+                    }
+                }
+            }
+        }
+        this.dataProvider = dataProvider
+        setNeedsDataRefresh()
+    }
+
+    @JvmStatic
+    fun crash() {
+        throw UnsupportedOperationException("This is a sample crash")
+    }
```

I see that this was in the original Java implementation as well, but it might be worth considering to remove it. I don't think it provides much context or value since we can simply throw an exception wherever we need it. If we did that, we could probably use a better message for it since we'd have more context about why we are crashing on purpose. What do you think?

---

I'm totally ok with removing it, I didn't understand its purpose in the first place but I assumed that it's useful for some cases I'm not aware of.

---

I agree that removing this seems like a good call.